x-pack/filebeat/input/gcs/config.go

```diff
@@ -56,34 +80,68 @@ type config struct {
+	// This value overrides the global BatchSize setting.
+	BatchSize *int `config:"batch_size"`
+	// MaxWorkers - Defines the maximum number of go routines that will be spawned.
+	// This value overrides the global MaxWorkers setting.
+	MaxWorkers *int `config:"max_workers" validate:"max=5000"`
+	// Poll - Defines if polling should be performed on the input bucket source.
+	// This value overrides the global Poll setting.
+	Poll *bool `config:"poll"`
+	// PollInterval - Defines the maximum amount of time to wait before polling for the next batch of objects from the bucket.
+	// This value overrides the global PollInterval setting.
+	PollInterval *time.Duration `config:"poll_interval"`
+	// ParseJSON - Informs the publisher whether to parse & objectify json data or not. By default this is set to
+	// false, since it can get expensive dealing with highly nested json data.
+	// This value overrides the global ParseJSON setting.
+	ParseJSON *bool `config:"parse_json"`
+	// FileSelectors - Defines a list of regex patterns that can be used to filter out objects from the bucket.
+	// This value overrides the global FileSelectors setting.
+	FileSelectors []fileSelectorConfig `config:"file_selectors"`
+	// ReaderConfig defines options for how object content is read and parsed for this specific bucket.
+	// This configuration overrides global ReaderConfig settings.
+	ReaderConfig readerConfig `config:",inline"`
+	// TimeStampEpoch - Defines the epoch time in seconds, which is used to filter out objects that are older than the specified timestamp.
+	// This value overrides the global TimeStampEpoch setting.
+	TimeStampEpoch *int64 `config:"timestamp_epoch"`
+	// ExpandEventListFromField - Defines the field name that will be used to expand the event into separate events.
+	// This is useful when the event is a list of events, and you want to expand it into separate events.
+	// This value overrides the global ExpandEventListFromField setting.
+	ExpandEventListFromField string `config:"expand_event_list_from_field"`
 }
 
-// fileSelectorConfig helps filter out gcs objects based on a regex pattern
+// fileSelectorConfig helps filter out GCS objects based on a regex pattern.
 type fileSelectorConfig struct {
+	// Regex is the regular expression pattern used to match object names.
 	Regex *match.Matcher `config:"regex" validate:"required"`
 	// TODO: Add support for reader config in future
 }
 
 // readerConfig defines the options for reading the content of an GCS object.
 type readerConfig struct {
-	Parsers  parser.Config `config:",inline"`
+	// Parsers contains the configuration for different content parsers (e.g., JSON, XML, CSV).
+	Parsers parser.Config `config:",inline"`
+	// Decoding specifies options for decoding the content, such as compression.
 	Decoding decoderConfig `config:"decoding"`
+	// ContentType suggests the MIME type of the object content, aiding in parsing.
```

Comment:
"suggests"? Is this intended to mean "hints"? If so, I'd use that instead since it's both shorter and already semantically coloured in this context.